polaris-migrator/src/migrations/replace-sass-z-index/replace-sass-z-index.ts

```diff
@@ -0,0 +1,119 @@
+import type {API, FileInfo, Options} from 'jscodeshift';
+import postcss, {Plugin} from 'postcss';
+import valueParser, {FunctionNode, Node} from 'postcss-value-parser';
+
+import {POLARIS_MIGRATOR_COMMENT} from '../../constants';
+
+interface PluginOptions extends Options {
+  namespace?: string;
+}
+
+const processed = Symbol('processed');
+
+function isNumericOperator(node: Node): boolean {
+  return (
+    node.value === '+' ||
+    node.value === '-' ||
+    node.value === '*' ||
+    node.value === '/' ||
+    node.value === '%'
+  );
+}
+
+const zIndexMap = {
+  content: '--p-z-1',
+  overlay: '--p-z-2',
+};
+
+const isValidElement = (
+  element: unknown,
+): element is keyof typeof zIndexMap => {
+  return Object.keys(zIndexMap).includes(element as string);
+};
+
+const hasMoreThanOneArgument = (node: FunctionNode) => {
+  if (!node.nodes || !Array.isArray(node.nodes)) return false;
+  return node.nodes && node.nodes.length > 1;
+};
+
+const plugin = (options: PluginOptions = {}): Plugin => {
+  const namespace = options?.namespace || '';
+  // This migration could be run over sass with
+  // proper legacy-polaris namespacing
+  // or some other pre-processed / post-processed css
+  // in which case we'll still want to transform a non-namespaced z-index fn.
+  const functionName = namespace ? `${namespace}.z-index` : 'z-index';
+  const isZIndexFn = (node: Node): node is FunctionNode => {
```

Comment:
We should replace this with `namespace()` and `isSassFunction` once those utilities make it into main from #7264

Reply:
Yes, I'll see if we can get that merged today so it will be available for other migrations 👍

Reply:
#7264 has been merged 🚢 those functions should now be ready for use

Reply:
Refactored the file to use these nice new utilities, ty 🎉